addons/metallb/addon.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+Pharos.addon 'metal-lb' do
+  version '0.7.3'
+  license 'Apache License 2.0'
+
+  config {
+    attribute :address_pools, Pharos::Types::Array
+    attribute :peers, Pharos::Types::Array
+  }
+
+  config_schema {
+    required(:address_pools).filled(min_size?: 1) do
+      each do
+        schema do
+          required(:name).filled(:str?)
+          required(:protocol).filled(:str?)
+          required(:addresses).each(:str?)
+        end
+      end
+    end
+
+    optional(:peers) do
+      each do
+        schema do
+          required(:peer_address).filled(:str?)
+          required(:peer_asn).filled(:int?)
+          required(:my_asn).filled(:int?)
+          optional(:node_selectors).each(:hash?)
+        end
+      end
+    end
+  }
+
+  install {
+    # Load the base stack
+    stack = kube_stack({})
+    puts stack.inspect
+    stack.resources << K8s::Resource.new(build_config)
+    stack.apply(kube_client)
+  }
+
+  def validate
+    super
+    # Validate BGP peers exist if BGP used
+    return unless config.address_pools.count { |pool| pool.dig(:protocol) == 'bgp' }.positive?
+    raise Pharos::InvalidAddonError, "Peers have to be configured for BGP protocol" if config.peers.nil? || config.peers.empty?
+  end
+
+  def build_config
+    {
+      apiVersion: 'v1',
+      kind: 'ConfigMap',
+      metadata: {
+        namespace: 'metallb-system',
+        name: 'config'
+      },
+      data: {
+        config: {
+          # This part is using string keys to get the output yaml correct as it's used as plain string
+          'address-pools' => config.address_pools.map { |pool|
+            {
+              'name' => pool[:name],
+              'protocol' => pool[:protocol],
+              'addresses' => pool[:addresses]
+            }
+          },
+          'peers' => config.peers.map { |peer|
```

Comment:
This should only be present if `config.address_pools.count { |pool| pool.dig(:protocol) == 'bgp' }.positive?`.